server/app/repository/ProgramRepository.java

```diff
@@ -438,6 +429,52 @@ private Query<ProgramModel> allProgramVersionsQuery(long programId) {
         .query();
   }
 
+  private ExpressionList<ApplicationModel> searchUsingPrimaryApplicantInfo(
+      String search, ExpressionList<ApplicationModel> query) {
+    // Remove all special characters
+    String maybeOnlyDigits = search.replaceAll("[^a-zA-Z0-9]", "");
+    // Check if remaining string is actually only digits
```

Comment:
If you can construct a trie using the query results before, you can fetch results faster.

Reply:
@swatkat1 can you say more about what you mean or provide an example?

Reply:
So when the page loads, query all the wellKnownpaths and create a Map<String,List<ApplicantIds>> searchMap. This will be your master list. Any time a search happens, directly query to the master list, and return the list of Applicants (from the values) as a search result.  We can talk offline too if interested.

Reply:
If I understand correctly, I think that would involve fetching data for all of the applicants at once (on page load), which I don't think we want to do because that would be a lot of data to hold in memory. Let me know if I'm misunderstanding though.

Is this a blocking request or more of a helpful suggestion?

Reply:
It is not a blocker but a suggestion. If you use all the Strings in the WellKnownPath to build a Trie, it will lesser space. But your call! I am ok with either.